src/http_tcp_server.cpp

```diff
@@ -115,29 +168,42 @@ namespace http
             exitWithError(ss.str());
         }
     }
-
-    std::string TcpServer::buildResponse()
-    {
-        std::string htmlFile = "<!DOCTYPE html><html lang=\"en\"><body><h1> HOME </h1><p> Hello from your Server :) </p></body></html>";
+    std::string TcpServer::buildHomepageResponse(bool include_content) {
+        std::ifstream html_file("../demo_content/index.html");
+        std::string html_string((std::istreambuf_iterator<char>(html_file)), std::istreambuf_iterator<char>());
         std::ostringstream ss;
-        ss << "HTTP/1.1 200 OK\nContent-Type: text/html\nContent-Length: " << htmlFile.size() << "\n\n"
-           << htmlFile;
-
-        return ss.str();
+        ss << "HTTP/1.1 200 OK\nContent-Type: text/html\nContent-Length: " <<html_string.size() << "\n\n" << html_string;
+        std::string response(ss.str());
+        if (include_content == true) {
+            response.append(html_string);
+        }
+        return response;
     }
-
-    void TcpServer::sendResponse()
-    {
+    std::string TcpServer::build404Response() {
+        std::ifstream html_file("../demo_content/404-page.html");
+        std::string html_string((std::istreambuf_iterator<char>(html_file)), std::istreambuf_iterator<char>());
+        std::ostringstream ss;
+        ss << "HTTP/1.1 404 Not Found\nContent-Type: text/html\nContent-Length: " <<html_string.size() << "\n\n" << html_string;
+        std::string response(ss.str());
+        response.append(html_string);
+        return response;
+    }
+    std::string TcpServer::build405Response() {
+        std::ifstream html_file("../demo_content/405-page.html");
+        std::string html_string((std::istreambuf_iterator<char>(html_file)), std::istreambuf_iterator<char>());
+        std::ostringstream ss;
+        ss << "HTTP/1.1 405 Method Not Allowed\nAllow: GET, HEAD "<< "\n\n" << html_string;
+        std::string response(ss.str());
+        response.append(html_string);
+        return response;
+    } 
```

Comment:
We had discussed hard-coding HTML pages for BCSW-11, so we wouldn't have to wait for BCSW-10.  I did go ahead and put the pages in their own HTML pages, but yes, there are still hard-coded functions for each page. I think we need to have a single response function that responds to all requests.

In the C world, we would create a struct, perhaps called response_info, and that struct would contain various parameters that the response handler would need in order to build the response (request type, status code, resource URI, maybe content-type and content-length.  The response handler would then receive a pointer to this struct and create the response accordingly.  Since this is C++, perhaps we should make a response class.